benchmarks/README.rst

```diff
@@ -16,22 +16,52 @@ unless told otherwise. Some of the benchmarking features in
 ``runtests.py``. To run the benchmarks, you do not need to install a
 development version of NumPy to your current Python environment.
 
-Run a benchmark against currently checked out NumPy version (don't
-record the result)::
+*Before beginning, ensure that* airspeed velocity *is installed.*
+By default, `asv` ships with support for anaconda and virtualenv::
+
+    pip install asv
+    pip install virtualenv
+
+After contributing new benchmarks, you should test them locally
+before submitting a pull request.
+
+To run all benchmarks, navigate to the root NumPy directory at
+the command line and execute::
+
+    python runtests.py --bench
+
+where ``--bench`` activates the benchmark suite instead of the
+test suite. This builds NumPy and runs the benchmarks.
+To run benchmarks from a particular benchmark module, such as
+``bench_core.py``, simply append the filename without the extension::
 
     python runtests.py --bench bench_core
 
-Compare change in benchmark results to another version::
+To run a benchmark defined in a class, such as ``Mandelbrot``
+from ``bench_avx.py``::
+
+    python runtests.py --bench bench_avx.Mandelbrot
+
+Compare change in benchmark results to another version/commit/branch::
 
     python runtests.py --bench-compare v1.6.2 bench_core
+    python runtests.py --bench-compare 8bf4e9b bench_core
+    python runtests.py --bench-compare master bench_core
 
-Run ASV commands (record results and generate HTML)::
+All of the commands above display the results in plain text in
+the console, and the results are not saved for comparison with
+future commits. For greater control, a graphical view, and to
+have results saved for future comparison you can run ASV commands
+(record results and generate HTML)::
 
     cd benchmarks
     asv run --skip-existing-commits --steps 10 ALL
```

Comment:
Consider changing this example too: this will run the entire benchmark suite 10x, i.e. ~2300 total benchmarks. This will take very long to run and as a result isn't very useful as an example. Consider replacing with an example that runs the benchmarking suite on the current commit:

```
cd benchmarks
# Run benchmarking suite with latest commit from current branch
asv run
asv publish
asv preview
```

Reply:
Modified the example to easy one.